pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1003,6 +1003,19 @@ func setMetricAnyValue(item *metrics.MetricItem, val interface{}) {
 func hardwareHealthTimerTask(ctx *zedagentContext, handleChannel chan interface{}) {
 	iteration := 0
 	log.Functionln("starting report health check timer task")
+
+	for deviceNetworkStatus != nil && deviceNetworkStatus.State != types.DPCStateSuccess && iteration < 100 {
+		log.Noticef("Waiting for device networking to be ready, Network Status: %s", deviceNetworkStatus.State)
+		time.Sleep(5 * time.Second)
+		iteration++
+	}
+
+	if iteration >= 100 {
+		log.Warnf("Device networking not ready after 500 seconds, giving up...")
+		return
+	}
```

Comment:
In this case shouldn't we still run the periodic report below?

An alternative solution is to get a success boolean from publishΗealthChecksReport() and start with a shorter timer (10 seconds or a minute) and when that function returns success switch to the HardwareHealthInterval timer.
That not only handles local failures like having no IP address, but also upstream network failures which happen on boot e.g., when all of the equipment in the machine/cabinet/truck (server, switches, routers) are powered on at the same time.

Reply:
The alternative we are suggesting sounds better. My only question is whether the condition "if deviceNetworkStatus != nil && deviceNetworkStatus.State != types. DPCStateSuccess" is sufficient for all cases.

Reply:
Such a check wouldn't handle the cases when the communication failure is further upstream. But you can avoid any such checks an instead have a self-adjusted timer which is shorter until it succeeds. We have something very similar in https://github.com/lf-edge/eve/blob/1ee884dd6e93e52ac6e4ea2577ab84b142697f0d/pkg/pillar/cmd/zedagent/handlecertconfig.go#L187